### PR TITLE
util/codec: encode int64 to varint if not comparable

### PR DESCRIPTION
**Pull request by coocood**
This will save a lot of storage space.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @BusyJay @qiuyesuifeng PTAL

**Comment by shenli** [comment]
Miss varint decode function?

File: util/codec/codec.go [hunk 1]
```diff
@@ -276,3 +306,19 @@ func peekDecimal(b []byte) (int, error) {
 	}
 	return l + bl, nil
 }
+
+func peekVarint(b []byte) (int, error) {
+	_, n := binary.Varint(b)
```
Comment (BusyJay) [inline util/codec/codec.go]: This may cause redundant scan.

Comment (coocood) [inline util/codec/codec.go]: If we don't scan, how do we get the varint size?

Comment (BusyJay) [inline util/codec/codec.go]: Return the decoded datum and the size directly.

Comment (coocood) [inline util/codec/codec.go]: This function is used in `CutOne`, which doesn't need the decoded datum.

Comment (BusyJay) [inline util/codec/codec.go]: Actually, the col id is always needed.

**Comment by shenli** [comment]
LGTM

File: util/codec/codec.go [hunk 2]
```diff
@@ -95,6 +91,28 @@ func encodeBytes(b []byte, v []byte, comparable bool) []byte {
 	return b
 }
 
+func encodeSignedInt(b []byte, v int64, comaprable bool) []byte {
+	if comaprable {
```
Comment (zimulala) [inline util/codec/codec.go]: comaprable -> comparable ?

**Comment by zimulala** [comment]
LGTM

**Pull request merged**